acton_agent/tools/openapi_tool.py

```
@@ -0,0 +1,618 @@
+        with open(spec, "r") as f:
+            content = f.read()
+
+        # Try JSON first
+        try:
+            return json.loads(content)
+        except json.JSONDecodeError:
+            # Try YAML
+            if yaml is None:
+                raise ValueError(
+                    "PyYAML is required to load YAML specs. Install with: pip install pyyaml"
+                )
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)